btc-price-fetcher.go

```diff
@@ -75,7 +108,30 @@ func fetchData() (CoinData, error) {
 	return coinData, nil
 }
 
-func writeDataToCSV(data CoinData, filename string) error {
+func fetchBinanceData() (BinanceData, error) {
+	uri := "https://api.binance.com/api/v3/ticker/price?symbol=BTCUSDT"
+
+	client := &http.Client{}
+
+	req, err := http.NewRequest("GET", uri, nil)
+	if err != nil {
+		fmt.Printf("Error: %v", err)
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println("Error sending the request to Binance API", err)
+	}
+
+	respBody, _ := io.ReadAll(resp.Body)
+	var coinData BinanceData
+	err = json.Unmarshal(respBody, &coinData)
+	if err != nil {
+		fmt.Println("Error unmarshalling JSON from Binance API: ", err)
+	}
+	return coinData, nil
+}
+
+func writeDataToCSV(data CoinData, binData BinanceData, filename string) error {
```

Comment:
task number 2, how to keep the three go routines running all the time
the coin fetch go routine sends updates to the writeData goroutine over channel